### PR TITLE
Add implementation of WindowPlayer for mac.

### PR DESCRIPTION
**Pull request by acely**
This is targeted at mac because Oracle's JDK do not have heavy weight
AWT components.And using CanvasVideoSurface may cause JVM crashing.
This commit provide a tricky way of getting AWT window's native id, and
make it easier for mac developers to use vlc.

### CODE REVIEW

**Comment by caprica** [comment]
I will need some time to review this.

On first inspection there is way too much "duplicate" code with existing vlcj classes. I would much prefer these changes to impact only the video surface adapter concept and not require a new separate media player. If that means changes to the video surface adapter concept to work with a Window rather than a Canvas specifically then so be it it.

**Comment by acely** [comment]
I've tried to reuse original code, but found that's specified with Canvas, doesn't work with Window. So I had to "duplicate" so much code to get it working.

**Comment by caprica** [comment]
So for me the biggest sticking point with this pull request was all the copy+paste duplication of the various classes.

As part of the vlcj-4 work that has recently started, I have changed the video surface component API so that it now takes a java.awt.Component rather than a Canvas. Of course, a Window is a Component too. These changes are relatively minor and you can now quite easily play videos in a Canvas (as before), a Window (as in this pull request), and even other components like a Button if you really want, with basically no application code changes. That much works just fine.

What that leaves is getting at least the java.awt.Window version working on OSX, using some trickery as you have mentioned to get the native window ID.

When I have finished the major vlcj4 structural changes, I will return to this and get it working on OSX.

In short, this pull request is no longer needed because of changes upcoming with vlcj-4.

In any case, thank you for this original pull request showing that this solution might just be possible on OSX - whilst not ideal, it may be good enough for some use-cases on OSX.

**Comment by acely** [comment]
Sure, that'll be great!
Proud of you!

**Comment by caprica** [comment]
I now have a working implementation on OSX that fits in very nicely with vlcj-4 API, without any code duplication, so I will close this pull request.

**Comment by caprica** [comment]
So... as is annoyingly typical with OSX there are some severe limitations.

The good news is that you definitely get a working media player with vlcj/LibVLC on modern JDKs on OSX using the approach discussed here.

The bad news is there is a significant limitation that I don't know how to solve right now...

Resizing the Window does not resize the native video - the native video rendering seems somehow locked to the original size of the window.

Did you encounter this when you developed this pull request and if so is there a nice solution?
